spec/unit/catalog_indexing_verifier_spec.rb

```diff
@@ -0,0 +1,136 @@
+# frozen_string_literal: true
+
+require "verifier/catalog_index_verifier"
+require "webmock"
+
+module PostZephirProcessing
+  RSpec.describe(CatalogIndexVerifier) do
+    let(:solr_url) { "http://solr-sdr-catalog:9033/solr/catalog" }
+
+    around(:each) do |example|
+      with_test_environment do
+        ClimateControl.modify(SOLR_URL: solr_url) do
+          example.run
+        end
+      end
+    end
+
+    def stub_catalog_record_count(result_count)
+      WebMock.enable!
+
+      url = "#{solr_url}/select?fq=deleted:false&q=*:*&rows=0&wt=json"
+
+      result = {
+        "responseHeader" => {
+          "status" => 0,
+          "QTime" => 0,
+          "params" => {
+            "q" => "*=>*",
+            "fq" => "deleted:false",
+            "rows" => "0",
+            "wt" => "json"
+          }
+        },
+        "response" => {"numFound" => result_count, "start" => 0, "docs" => []}
+      }.to_json
+
+      WebMock::API.stub_request(:get, url)
+        .to_return(body: result, headers: {"Content-Type" => "application/json"})
+    end
+
+    def stub_catalog_timerange(date, result_count)
+      # must be like YYYY-mm-ddTHH:MM:SSZ - iso8601 with a 'Z' for time zone -
+      # time zone offsets like DateTime.iso8601 produce by default are't
+      # allowed for solr
+
+      # FIXME: don't love that we duplicate this logic & the URL between here &
```

Comment:
Rethink -- test that things were indexed since the file was produced. The timestamp on the file might not be accurate (at the file system level). What do we want to assume as the actual _time_ for a given date?

I think that given a file e.g. `zephir_upd_20241212.json.gz` things in it should be indexed **after midnight eastern the following day.**

After midnight UTC the following day is fine, just not as strong a guarantee as it could be.